_posts/2022-08-05-hacking-for-cruise-control-ui.md

```diff
@@ -0,0 +1,300 @@
+---
+layout: post
+title: "Hacking Strimzi for Cruise Control UI"
+date: 2022-10-31
+author: kyle_liberti
+---
+
+For some time now, Strimzi has used [Cruise Control](https://github.com/linkedin/cruise-control) for automated partition rebalancing and has offered Strimzi custom resources for safely configuring, deploying, and communicating with Cruise Control through the Strimzi Operator.
+These custom resources provide great support for declaritive rebalancing.
+That being said, rather than working with custom resources, you might prefer to interact with applications like Cruise Control through a graphical user interface.
+LinkedIn provides such an interface with their [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) application which supports several Cruise Control operations like viewing Kafka cluster status and executing partition rebalances, all with just a couple button clicks.
+Although Strimzi does not offer direct support for the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) it can still be run with and connected to a Strimzi-managed Cruise Control instance.
+
+This post will walk through how to do so! 
+
+**Warning**
+
+Setting up the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) as described in this blog post requires disabling Cruise Control REST API settings.
+Having these security settings disabled exposes the Cruise Control API to potentially destructive Cruise Control operations that are not coordinated or supported by Strimzi Operator.
```

Comment:
Do you mean users could manually trigger a rebalance that could interfere with the Operator?
It would be good to clarify so users know what not to do.

Reply:
You make it sound very dangerous. I wonder if we could say something like:
```
Without authentication, anybody with access to the Cruise Control API can use all the available endpoints and trigger administrative actions such as rebalancing load or decommissioning brokers  for example.
```